contents/tutorials/web-worker.md

```diff
@@ -0,0 +1,221 @@
+    
+    posthog.init(
+      '<ph_project_api_key>', 
+      { 
+        api_host: '<ph_client_api_host>',
+        advanced_disable_decide: true
+      }
+    );
+  </script>
+</head>
+<body>
+  <button onclick="startTask()">Start Task</button>
+  <div id="result"></div>
+
+  <script>
+    function handleWorkerMessages() {
+      const worker = new Worker('worker.js');
+      
+      worker.onmessage = function(e) {
+        document.getElementById('result').textContent = `Result received: ${e.data}`;
+      };
+
+      return worker;
+    }
+
+      function startTask() {
+        const worker = handleWorkerMessages();
+        worker.postMessage('start');
+      }
+  </script>
+</body>
+</html>
+```
+
+> **Note:** We disable `advanced_disable_decide` as it causes a CORS error on our local server. This isn't the case in production so you shouldn't do this for a production-ready app.
+
+## Making PostHog work with the web worker
+
+To make PostHog work with a web worker, we can set up a new `posthog_event` message in `worker.js` that includes the event name and properties.
+
+```js
+// worker.js
+onmessage = function(e) {
+  if (e.data === 'start') {
+    let result = performWork();
+    
+    // Send PostHog event back to main thread
+    postMessage({
+      type: 'posthog_event',
+      eventName: 'worker_task_completed',
+      properties: {
+        result: result,
+        duration_ms: 1000
+      }
+    });
+    
+    // Send the result back to the main thread
+    postMessage(result);
+  }
+};
+
+// ... rest of your existing code
+```
+
+We can then handle the `posthog_event` message in `index.html` with our web PostHog initialization.
+
+```html
```

Comment:
Looks like this is a large code block for `index.html` where the content is almost all the same as the one above. I wonder if it should just show the function that changed?

Actually, I wonder if our Markdown parser supports `diff` as the language like GitHub does:
https://stackoverflow.com/a/40883538

Reply:
^this tbh